dotnet/src/Agents/OpenAI/Internal/AssistantRunOptionsFactory.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Collections.Generic;
+using OpenAI.Assistants;
+
+namespace Microsoft.SemanticKernel.Agents.OpenAI.Internal;
+
+/// <summary>
+/// Factory for creating <see cref="RunCreationOptions"/> definition.
+/// </summary>
+/// <remarks>
+/// Improves testability.
+/// </remarks>
+internal static class AssistantRunOptionsFactory
+{
+    /// <summary>
+    /// Produce <see cref="RunCreationOptions"/> by reconciling <see cref="OpenAIAssistantDefinition"/> and <see cref="OpenAIAssistantInvocationOptions"/>.
+    /// </summary>
+    /// <param name="definition">The assistant definition</param>
+    /// <param name="invocationOptions">The run specific options</param>
+    public static RunCreationOptions GenerateOptions(OpenAIAssistantDefinition definition, OpenAIAssistantInvocationOptions? invocationOptions)
+    {
+        int? truncationMessageCount = ResolveExecutionSetting(invocationOptions?.TruncationMessageCount, definition.ExecutionOptions?.TruncationMessageCount);
+
+        RunCreationOptions options =
+            new()
+            {
+                MaxCompletionTokens = ResolveExecutionSetting(invocationOptions?.MaxCompletionTokens, definition.ExecutionOptions?.MaxCompletionTokens),
+                MaxPromptTokens = ResolveExecutionSetting(invocationOptions?.MaxPromptTokens, definition.ExecutionOptions?.MaxPromptTokens),
+                ModelOverride = invocationOptions?.ModelName,
+                NucleusSamplingFactor = ResolveExecutionSetting(invocationOptions?.TopP, definition.TopP),
+                ParallelToolCallsEnabled = ResolveExecutionSetting(invocationOptions?.ParallelToolCallsEnabled, definition.ExecutionOptions?.ParallelToolCallsEnabled),
+                ResponseFormat = ResolveExecutionSetting(invocationOptions?.EnableJsonResponse, definition.EnableJsonResponse) ?? false ? AssistantResponseFormat.JsonObject : null,
+                Temperature = ResolveExecutionSetting(invocationOptions?.Temperature, definition.Temperature),
+                //ToolConstraint - Not Currently Supported (https://github.com/microsoft/semantic-kernel/issues/6795)
```

Comment:
Removed commented out code?

Reply:
This is linked to an issue in our backlog (see comment) that we will action on in the future once the `FunctionChoice` work is available.